File: src/main/java/net/tianben/advancedcombat/AdvancedCombat.java
package net.tianben.advancedcombat;

import net.fabricmc.api.ModInitializer;

import net.tianben.advancedcombat.block.ModBlocks;
import net.tianben.advancedcombat.item.group.ModItemGroups;
import net.tianben.advancedcombat.item.ModItems;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class AdvancedCombat implements ModInitializer {
	public static final String MOD_ID = "advancedcombat";
    public static final Logger LOGGER = LoggerFactory.getLogger(MOD_ID);

	@Override
	public void onInitialize() {
		ModItemGroups.registerItemGroups();

		ModBlocks.registerModBlocks();
		ModItems.registerModItems();
	}
}

File: src/main/kotlin/net/tianben/advancedcombat/AdvancedCombat.kt
package net.tianben.advancedcombat

import net.fabricmc.api.ModInitializer
import net.minecraft.util.Identifier
import net.tianben.advancedcombat.block.ModBlocks
import net.tianben.advancedcombat.item.ModItems
import net.tianben.advancedcombat.item.group.ModItemGroups
import net.tianben.advancedcombat.recipes.ModRecipes
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class AdvancedCombat : ModInitializer {

    companion object {
        const val MOD_ID: String = "advancedcombat"
        val LOGGER: Logger? = LoggerFactory.getLogger(MOD_ID)

        fun ID(path: String): Identifier {
            return Identifier(MOD_ID, path)
        }
    }

    override fun onInitialize() {
        ModItemGroups.registerItemGroups()

        ModBlocks.registerModBlocks()
        ModItems.registerModItems()

        ModRecipes.registerModRecipesSerializers()
    }
}